client/layout/main-banner/index.js

```diff
@@ -0,0 +1,31 @@
+import { Gridicon } from '@automattic/components';
+import './style.scss';
+
+export function MainBanner( { sectionName, siteIntent, siteSlug } ) {
+	const allowList = [ 'posts', 'media', 'pages', 'home' ];
```

Comment:
Micro-optimization nit: sounds like this constant be declared outside of the function. There's no need for us to redeclare it on every re-render.